player.gd

```diff
@@ -54,13 +57,19 @@ func _process(delta):
 	# When player hits bottom, cancel all speed
 	if position.y == screen_size.y:
 		speed = 0
-
-	if velocity.y > 0 && not in_air:
+	else:
+		speed = clamp(speed, -MAX_SPEED, MAX_SPEED)
+	
+	# Sprite animation based on direction of player input (neutral in air)
+	if direction < 0 && not in_air:
 		$AnimatedSprite2D.animation = "swim_down"
-	elif velocity.y < 0 && not in_air:
+	elif direction > 0 && not in_air:
 		$AnimatedSprite2D.animation = "swim_up"
 	else:
 		$AnimatedSprite2D.animation = "swim_neutral"
+		
+	# Rotation based on velocity
+	rotation = -PI / 2 * speed / (MAX_SPEED * 1.5)
```

Comment:
Can you explain this formula? I specifically don't get why you divide speed by max*1.5

Reply:
Without the 1.5, the rotation would end up directly vertical (up or down), so multiplying MAX_SPEED by 1.5 means that the rotation will be between 2/3 and -2/3 from x axis (rotation would be between 60 and -60 degrees). 

Rather than doing a clamp on rotation between the hardcoded radian values for those angles, I figured making the reduced rotation occur over a small distance would let it be slower, i.e. smoother.

Reply:
so you're doing pi/3 to get a 120deg cone, basically?

I still think the angle/rotation should be set by creating a vector that uses the fish's y speed and the obstacle's x speed, since that's the apparent velocity of the fish